code/lib/telemetry/src/package-json.ts

```diff
@@ -20,7 +21,17 @@ export const getActualPackageVersion = async (packageName: string) => {
 };
 
 export const getActualPackageJson = async (packageName: string) => {
-  // eslint-disable-next-line import/no-dynamic-require,global-require
-  const packageJson = require(path.join(packageName, 'package.json'));
+  const resolvedPackageJson = require.resolve(path.join(packageName, 'package.json'), {
+    paths: [process.cwd()],
+  });
+  const packageJson = await readJson(resolvedPackageJson);
   return packageJson;
 };
+
+// Note that this probably doesn't work in PNPM mode
+export const getStorybookCoreVersion = async () => {
+  const coreVersions = await Promise.all(
+    ['@storybook/core-common', '@storybook/core-server'].map(getActualPackageVersion)
+  );
+  return coreVersions.find((v) => v.version)?.version;
+};
```

Comment:
Can we reuse the code we use to check all SB versions are in sync for this (i.e. check all direct dependencies that are from the monorepo) rather than just trying a couple of packages that won't work in pnpm?

Reply:
That code needs a lot of work. I'll leave this for now, but we should definitely invest in making this area more robust.